src/shims/lock.h

```diff
@@ -285,6 +285,9 @@ _dispatch_thread_event_init(dispatch_thread_event_t dte)
 	dte->dte_value = 0;
 #else
 	_dispatch_sema4_init(&dte->dte_sema, _DSEMA4_POLICY_FIFO);
+#if defined(_WIN32)
+	_dispatch_sema4_create(&dte->dte_sema, _DSEMA4_POLICY_FIFO);
```

Comment:
this seems wrong, the point of the sema4 is to be lazy and do the create() only if we're hitting the kernel
why can't windows do the same?

Reply:
Or alternatively, you should make `_dispatch_sema4_init` always create, which is how the posix semaphore works (`_dispatch_sema4_is_created` always says `true`).

Reply:
Unfortunately I don't remember the test (it was in XCTest's test suite), but the queue wouldn't drain with the lazy semaphore.  I've adopted the posix semaphore policy instead as you suggested.  I think that it would be a good thing to do as a follow up (investigate and fix the lazy semaphores).